tests/includes/juju.sh

```diff
@@ -289,6 +317,9 @@ post_add_model() {
 		add_images_for_vsphere
 		;;
 	esac
+
+	arch=$(uname -m)
```

Comment:
we shouldn't be setting arch based on the machine running the test
arch needs to be explicitly passed in as a test parameter; if we say we're testing on arm64, then we pass that as a test arg and explicitly use it when deploying etc